adding std_dev calculation

MichaelClifford opened the pull request:
initial commit.

Commit:
initial change

MichaelClifford commented on flatliners/stddevcluster.py:
```diff
@@ -34,9 +35,32 @@ def calculate_mean(values, previous = None):
+    def calculate_stdv(values, previous = None):
+        if previous:
+
+            total = previous['total'] + int(values[0][0]) # I'm not 100% sure if this is the correct way to organize the data.
```
I'm a little unclear as to what is being passed as the `values` parameter. From inspection it looks like a list of multiple samples, where each sample is a list, with position [0] being some large float that represents the resource type (?) and position [1]  representing the actual value?

durandom replied:
It's the [metric_values](https://github.com/AICoE/prometheus-flatliner/blob/master/flatliners/baseflatliner.py#L33)
That's the array of `[timestamp, value]` pairs.

Actually, we could create another metrics processor that flattens these arrays. So you dont have to this in every processor.

Commit:
add comments and std_dev calculation

MichaelClifford commented:
Added a second commit that adds some comments to the code to help clarify what's happening at each step of app.py and stddevcluster.py `on_next`. 

Also updated `calculate_stdv` so that it should calculate the correct on-line standard deviation value for each step. I've also included the resource name and cluster_id in the output just as a sanity check.

durandom commented:
If you feel this is ready for review or merge, let us know and remove the WIP tag

MichaelClifford commented:
@durandom Please review. 

One thing to note, I opened issues #8 which I'll work on this morning to confirm the output here is the same as performing the std_dev calculation on all data at once for each resource .

durandom commented on flatliners/stddevcluster.py:
```diff
@@ -34,9 +42,55 @@ def calculate_mean(values, previous = None):
+    @staticmethod
+    def calculate_stdv(values, name, cluster, previous = None):
+        num_entires = len(values)
+        if previous:
```
This huge `if/else` block would be a good candidate to refactor into 2 methods

durandom reviewed (approved):
just the minor refactoring nitpick and all these `line too long` comments.
I'll look into increasing the allowed line length.

Other than that. Great stuff.
I'll merge

🎉
